File: languages/javascript/gladiator.js
const fs = require('fs');
const s = fs.readFileSync(0, 'utf8').toString();
let out = '';
if (s.indexOf("\"discipline\":\"construct\"") >= 0) {
  if (s.indexOf("\"arena\":\"cheeks\"") >= 0) out = "()";
  else if (s.indexOf("\"arena\":\"ears\"") >= 0) out = "/\\\\/\\\\";
  else if (s.indexOf("\"arena\":\"eyes\"") >= 0) out = "oo";
  else if (s.indexOf("\"arena\":\"geometry\"") >= 0) out = "";
  else if (s.indexOf("\"arena\":\"head_top\"") >= 0) out = "_";
  else if (s.indexOf("\"arena\":\"mouth\"") >= 0) out = "^";
  else if (s.indexOf("\"arena\":\"newlines\"") >= 0) out = "";
  else if (s.indexOf("\"arena\":\"nose\"") >= 0) out = ".";
  else if (s.indexOf("\"arena\":\"padding\"") >= 0) out = "      ";
  else if (s.indexOf("\"arena\":\"whiskers\"") >= 0) out = "><";
  process.stdout.write('{"ok":true,"output":"' + out + '"}');
} else {
  const ok = s.indexOf("\"candidate\":\" /\\\\_/\\\\\\n( o.o )\\n > ^ <\"") >= 0;
  process.stdout.write('{"ok":' + (ok ? 'true' : 'false') + ',"output":"' + (ok ? "es un gato" : "no es el gato canonico") + '"}');
}

File: languages/haskell/gladiator.hs
import Data.List (isInfixOf)
main :: IO ()
main = do
  s <- getLine
  if isInfixOf "\"discipline\":\"construct\"" s
    then putStrLn ("{\"ok\":true,\"output\":\"" ++ piece s ++ "\"}")
    else
      let ok = isInfixOf "\"candidate\":\" /\\\\_/\\\\\\n( o.o )\\n > ^ <\"" s
          v = if ok then "es un gato" else "no es el gato canonico"
      in putStrLn ("{\"ok\":" ++ (if ok then "true" else "false") ++ ",\"output\":\"" ++ v ++ "\"}")
piece :: String -> String
piece s = case [v | (k, v) <- table, isInfixOf k s] of
  (v:_) -> v
  []    -> ""
table :: [(String, String)]
table = [
      ("\"arena\":\"cheeks\"", "()"),
      ("\"arena\":\"ears\"", "/\\\\/\\\\"),
      ("\"arena\":\"eyes\"", "oo"),
      ("\"arena\":\"geometry\"", ""),
      ("\"arena\":\"head_top\"", "_"),
      ("\"arena\":\"mouth\"", "^"),
      ("\"arena\":\"newlines\"", ""),
      ("\"arena\":\"nose\"", "."),
      ("\"arena\":\"padding\"", "      "),
      ("\"arena\":\"whiskers\"", "><"),
    ("", "")
  ]
